reproject/interpolation/core_celestial.py

```diff
@@ -48,12 +51,20 @@ def _reproject_celestial(array, wcs_in, wcs_out, shape_out, order=1):
     # remainder of the array. We then operate on the view, but this will change
     # the original array with the correct shape.
 
-    array_new = np.zeros(shape_out)
+    if array_out is not None:
+        if tuple(array_out.shape) != tuple(shape_out):
```

Comment:
`shape_out` was a list when I tried it.  Tupling the `array_out.shape` was overkill, and I can remove that, but I think it is possible that a user would pass a list, since our [own code](https://github.com/astrofrog/reproject/blob/master/reproject/utils.py#L36) does.